internal/service/backupcompliancepolicy/resource_backup_compliance_policy_test.go

```diff
@@ -419,3 +455,48 @@ func basicChecks() []resource.TestCheckFunc {
 	checks = append(checks, checkExists(resourceName), checkExists(dataSourceName))
 	return checks
 }
+
+func configBasicWithOptionalAttributesWithNonDefaultValues(projectName, orgID, projectOwnerID string) string {
+	return acc.ConfigProjectWithSettings(projectName, orgID, projectOwnerID, false) +
+		`resource "mongodbatlas_backup_compliance_policy" "backup_policy_res" {
+		project_id                 = mongodbatlas_project.test.id
+		authorized_email           = "test@example.com"
+		authorized_user_first_name = "First"
+		authorized_user_last_name  = "Last"
+		copy_protection_enabled    = true
+		pit_enabled                = false
+		encryption_at_rest_enabled = false
+		
+		restore_window_days = 7
+		
+		on_demand_policy_item {
+			frequency_interval = 0
+			retention_unit     = "days"
+			retention_value    = 3
+		}
+		
+		policy_item_hourly {
+			frequency_interval = 6
+			retention_unit     = "days"
+			retention_value    = 7
+		}
+		
+		policy_item_daily {
+			frequency_interval = 0
+			retention_unit     = "days"
+			retention_value    = 7
+		}
+		
+		policy_item_weekly {
+			frequency_interval = 0
+			retention_unit     = "weeks"
+			retention_value    = 4
+		}
+		
+		policy_item_monthly {
```

Comment:
do we support `retention_unit = "year"` also?

Reply:
yearly policy is tested in other tests. In this test I just want to test that there are no plan changes when the configuration sets non default values to an optional attribute